lib/uplinkc/caveat.go

```diff
@@ -0,0 +1,26 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package main
+
+// #include "uplink_definitions.h"
+import "C"
+
+import "storj.io/storj/pkg/macaroon"
+
+// export new_caveat
+// new_caveat returns a Caveat with a random generated nonce.
+func new_caveat(cerr **C.char) C.Caveat {
+	caveat, err := macaroon.NewCaveat()
+	if err != nil {
+		*cerr = C.CString(err.Error())
```

Comment:
Do we want to return immediately with this error?

Reply:
returning now immediately